compiler/noirc_frontend/src/hir/comptime/value.rs

```diff
@@ -502,19 +504,35 @@ impl Value {
         Ok(vec![token])
     }
 
-    /// Converts any unsigned `Value` into a `u128`.
-    /// Returns `None` for negative integers.
-    pub(crate) fn to_u128(&self) -> Option<u128> {
+    /// Returns false for non-integral `Value`s.
+    pub(crate) fn is_integral(&self) -> bool {
+        matches!(
+            self,
+            Self::Field(_)
+                | Self::I8(_)
+                | Self::I16(_)
+                | Self::I32(_)
+                | Self::I64(_)
+                | Self::U8(_)
+                | Self::U16(_)
+                | Self::U32(_)
+                | Self::U64(_)
+        )
```

Comment:
Can you tighten up the formatting here? Maybe importing the Value variants would format this onto fewer lines.

Reply:
Ok, updated. I could also split it into `is_signed_integer` / `is_unsigned_integer` or similar